entrypoint.sh

```diff
@@ -22,18 +22,39 @@ MESSAGE="${MESSAGE:-}"
 NAME=$(jq ".pusher.name" "$GITHUB_EVENT_PATH")
 EMAIL=$(jq ".pusher.email" "$GITHUB_EVENT_PATH")
 
+[[ "$GITHUB_EVENT_NAME" == "delete" ]] && event_ref=$(jq --raw-output .ref "$GITHUB_EVENT_PATH") || event_ref=""
```

Comment:
Yeah so I was contemplating whether to include this as it's a bit specific to my use case. Basically on a `delete` event I want to know the name of the branch/tag/ref that got deleted. The `GITHUB_REF` that is set by default seems to point to master. The conditional setting is due to the path to the `.ref` being different for different event types. 

I could probably make this a little better and able to parse the different objects based on the event type? Although I'm not sure if there are any other use cases where this info would be needed.

Reply:
Ah, I see now!

Why don't we pass set the following environment variables, all the time, without the delete conditional?

```bash
GITHUB_EVENT_NAME="$GITHUB_EVENT_NAME"
GITHUB_EVENT_REF=$(jq ".ref" "$GITHUB_EVENT_PATH")
```

Reply:
> GITHUB_EVENT_REF=$(jq ".ref" "$GITHUB_EVENT_PATH")
The path to the ref is different depending on the event. For example for a pull request it's:
```
.pull_request.<head|base>.ref
```
Push and delete are the same, haven't dug through the other events. I guess we could make it work for those 3 types, leave it unset for others. Probably handles the majority of use cases?

Reply:
I see! What events does an action receive, when you delete a branch/tag? Do you also get a push event, with null `after` field, ala the example payload in https://developer.github.com/v3/activity/events/types/#pushevent?

Reply:
Not sure, will need to go back and check. All I know is that the default `GITHUB_REF` passed in on a delete event was pointing at master and not what was deleted.

Reply:
So I suspect the action only gets the event you configured in your workflow, which in my case is always a delete event. 

In order to make this more flexible, I might add the ability to handle any type of event and parse the ref from it, sound reasonable?

Reply:
Yeah sounds reasonable to me! Maybe a separate PR/release thought? 🤷🏼‍♂️

Reply:
Do you want me to leave what's there now? i.e. just catering for delete events?

Reply:
Ah sorry, I somehow thought we removed the delete ref check.

I reckon either we remove the GitHub-based env vars (and just have this PR add the custom env vars), or we support all the events from the get-go (which I know is a bit of work to do sorry! Hence my suggestion for splitting up the PR).

What do you reckon?

Reply:
Agreed. I'll pull the ref check out and work on a separate PR mapping all of the event types.

Reply:
So, looking at this https://developer.github.com/actions/managing-workflows/workflow-configuration-options/#events-supported-in-workflow-files

For most events, the default value of `GITHUB_REF` is probably fine, it seems to be the information you'd want for the event in question. It's only for delete where the `GITHUB_REF` is set to `default branch` but I want to know the name of the branch that was deleted. I might need to brainstorm a way to do this, because I feel that I'll end up writing a big case statement to do nothing in all cases but one :P

Reply:
Okay! Thanks for the digging. Why don’t we just always set the `GITHUB_REF` and pass through what Actions gives, so it matches the default behavior, and add a `GITHUB_DELETED_REF` just for the delete action for this extra piece of info you want? That way it’s not “this thing is sometimes this and sometimes that” but “This thing is always this, and for certain events we set these extra env vars”?

Reply:
Yeah sounds good. BTW you are already always passing `GITHUB_REF` as the default value for the `BRANCH` arg. I'll leave that as is, but open a new PR with the proposed addition of `GITHUB_DELETED_REF`

Reply:
Yep cool! I think if you're manually settings the branch/commit of the pipeline you're running, that's exactly the sort of time you probably want to know the original info… the pipeline that's running could be completely separate from the Action repo that triggered it.

Reply:
Cool. Well in that case this PR is ready for your review. Let me know if you want anything else changed.